### PR TITLE
Wigi: make justify in JS WordSpacing-based

### PR DESCRIPTION
**Pull request by ykuzmenko73**
Try to make Justify and join-text optimization work together
Under flag js_justify it uses WordSpacing css style instead of arranging words manually

### CODE REVIEW

**Comment by ykuzmenko73** [comment]
[Reader works slow in Safari if text is JUSTIFIED](https://trello.com/c/V5Spps48/16755-reader-works-slow-in-safari-if-text-is-justified)

File: lib/form/paragraph3.flow [hunk 1]
```diff
@@ -488,8 +488,9 @@ getParaLineInfos(
 		} else 1.0
 	} else 1.0;
 	reflowResult = reflowParaWords(wordGroups, availableWidth, indent, rtl, fontSize, textScale);
-	map(reflowResult, \line -> {
-		optimizedLine = getOptimizedLine(line.words, alignment);
+	lines = length(reflowResult) - 1;
```
Comment (brawaga) [inline lib/form/paragraph3.flow:491]: linesCount maybe better.

Comment (ykuzmenko73) [reply lib/form/paragraph3.flow]: renamed

File: lib/form/paragraph3.flow [hunk 2]
```diff
@@ -953,7 +954,7 @@ renderParaLines(
 		// On the last line, we do not justify.
 		// Also if the line is single with TightWidth, any alignment is equal to StartAlign
 		lineAlignment =
-			if (alignment == Justify() && lastLine) {
+			if (alignment == Justify() && lastLine && !newJustifyfyRenderEnabled) {
```
Comment (brawaga) [inline lib/form/paragraph3.flow:957]: «fyfy» typo here.

**Review by brawaga — APPROVED** [review]
LGTM

**Pull request merged**